src/megatron/energon/wrappers/repeat_dataset.py

```diff
@@ -64,8 +68,22 @@ def __iter__(self) -> Iterator[T_sample]:
             self.repeats is not None or self.dataset.worker_has_samples()
         ), "Cannot repeat empty dataset indefinitely"
         while self.repeats is None or self._repetition[worker_idx] < self.repeats:
+            ds_len = len(self.dataset)
```

Comment:
This should remain constant, but may involve computation. Move outside of the loop.